src/app/(sidebar)/transaction/sign/components/Import.tsx

```diff
@@ -0,0 +1,116 @@
+"use client";
+
+import { useState } from "react";
+import { Alert, Card, Icon, Text, Button } from "@stellar/design-system";
+import { TransactionBuilder } from "@stellar/stellar-sdk";
+
+import { useStore } from "@/store/useStore";
+
+import { validate } from "@/validate";
+
+import { XdrPicker } from "@/components/FormElements/XdrPicker";
+
+export const Import = () => {
+  const { network, transaction } = useStore();
+  const { updateSignActiveView, updateSignImportXdr, updateSignImportTx } =
+    transaction;
+
+  const [txXdr, setTxXdr] = useState<string>("");
+  const [txErrMsg, setTxErrMsg] = useState<string>("");
+  const [txSuccessMsg, setTxSuccessMsg] = useState<string>("");
+
+  // on textarea onChange for 'import tx xdr'
+  const onChange = (value: string) => {
+    // reset messages onChange
+    setTxErrMsg("");
+    setTxSuccessMsg("");
+    setTxXdr(value);
+
+    if (value.length > 0) {
+      const validatedXDR = validate.xdr(value);
+
+      if (validatedXDR.result === "success") {
+        setTxSuccessMsg(validatedXDR.message);
+      } else {
+        setTxErrMsg(validatedXDR.message);
+      }
+    }
+  };
+
+  const onImport = () => {
+    try {
+      const transaction = TransactionBuilder.fromXDR(txXdr, network.passphrase);
+
+      updateSignImportTx(transaction);
+      updateSignImportXdr(txXdr);
+
+      // change to 'overview' view when successfully imported
+      updateSignActiveView("overview");
+    } catch (e) {
+      setTxErrMsg("Unable to import a transaction envelope");
+      updateSignActiveView("import");
+    }
+  };
+
+  return (
```

Comment:
We're not persisting entered data when the page is refreshed. Is that intentional?

Reply:
that's how it is in the current lab

Reply:
The current Lab's state is broken, so it's not persisting the data after the page refresh. It would be a better UX if it did. What do you think?

Reply:
task created here https://github.com/stellar/laboratory/issues/845